### PR TITLE
Rename TransferMode to BlendMode

### LINKED ISSUE #7200: Renamed TransferMode to BlendMode
Skia calls this BlendMode now and that's a better name.

### PR DESCRIPTION
**Pull request by abarth**
Skia calls this BlendMode now and that's a better name.

Fixes #7200

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @chinmaygarde

File: packages/flutter/lib/src/rendering/proxy_box.dart [hunk 1]
```diff
@@ -800,17 +800,17 @@ class RenderShaderMask extends RenderProxyBox {
     markNeedsPaint();
   }
 
-  /// The [TransferMode] to use when applying the shader to the child.
+  /// The [BlendMode] to use when applying the shader to the child.
   ///
-  /// The default, [TransferMode.modulate], is useful for applying an alpha blend
+  /// The default, [BlendMode.modulate], is useful for applying an alpha blend
   /// to the child. Other transfer modes can be used to create other effects.
```
Comment (chinmaygarde) [inline packages/flutter/lib/src/rendering/proxy_box.dart]: Doc update: "Other blend modes..."

File: packages/flutter/lib/src/widgets/basic.dart [hunk 2]
```diff
@@ -129,25 +129,25 @@ class ShaderMask extends SingleChildRenderObjectWidget {
   /// it can customize the shader to the size and location of the child.
   final ShaderCallback shaderCallback;
 
-  /// The [TransferMode] to use when applying the shader to the child.
+  /// The [BlendMode] to use when applying the shader to the child.
   ///
-  /// The default, [TransferMode.modulate], is useful for applying an alpha blend
+  /// The default, [BlendMode.modulate], is useful for applying an alpha blend
   /// to the child. Other transfer modes can be used to create other effects.
```
Comment (chinmaygarde) [inline packages/flutter/lib/src/widgets/basic.dart]: Same: Docs need to be updated. "Other blend modes..."

**Review by chinmaygarde — APPROVED** [review]
lgtm

**Comment by Hixie** [comment]
LGTM

**Pull request merged**